### ISSUE TITLE
Regression in ref resolution for local nesting

### ISSUE THREAD

**Issue by godric** [davishmcclurg/json_schemer#131]
# What?
Somewhere between 0.2.25 and 1.0.0 the resolver stopped working for in-document nested reference from a different schema (see reproduction bellow).

# Reproduction:
```ruby
require 'json_schemer'

SCHEMAS = {
  'one' => {
    "$schema" => "http://json-schema.org/draft-07/schema#",
    "type" => "object",
    "properties" => {
      "value" => {
        "$ref" => "#/definitions/ref_nested"
      }
    },
    "definitions" => {
      "ref_nested" => {
        "$ref" => "#/definitions/ref_bare"
      },
      "ref_bare" => {
        "type" => "string"
      }
    }
  },
  'two' => {
    "$schema" => "http://json-schema.org/draft-07/schema#",
    "properties" => {
      "value" => {
        "$ref" => "one#/definitions/ref_bare"
      }
    },
  },
  'three' => {
    "$schema" => "http://json-schema.org/draft-07/schema#",
    "properties" => {
      "value" => {
        "$ref" => "one#/definitions/ref_nested"
      }
    },
  }
}

ref_resolver = ->(uri) { SCHEMAS[URI(uri).path] }

schema_one = JSONSchemer.schema(SCHEMAS['one'], ref_resolver: ref_resolver)
schema_two = JSONSchemer.schema(SCHEMAS['two'], ref_resolver: ref_resolver)
schema_three = JSONSchemer.schema(SCHEMAS['three'], ref_resolver: ref_resolver)

schema_one.valid?({ "value" => "" })
# true (works)

schema_two.valid?({ "value" => "" })
# true (works)

schema_three.valid?({ "value" => "" })
# Exception: JSONSchemer::InvalidRefResolution (broken)
```

Let me know if I should provide more infos and/or can help in any way.
PS: Really appreciate the work and recent updates to this Gem 🙇

**Comment by davishmcclurg** [comment]
Thanks for the issue and reproducible case. I think I've got a fix here: https://github.com/davishmcclurg/json_schemer/compare/json-pointer-ref

Do you mind checking it out to make sure it works for your use cases?

**Comment by godric** [comment]
So fast!
I think you've got it, the change does seem to solve the issue. I validated a large number of objects against my deeply nested schema and everything works as expected.

Looking forward to seeing it in a future release of the gem 👍

**Comment by davishmcclurg** [comment]
Thanks for testing it out @godric! I released that fix in v1.0.3.
